test/integration/helpers_test.go

```diff
@@ -416,3 +417,11 @@ func setIngressClassNameWithRetry(ctx context.Context, namespace string, obj run
 	}
 	return fmt.Errorf("unsupported GroupVersionKind %v", obj.GetObjectKind())
 }
+
+func skipTestForExpressionRouter(t *testing.T) {
```

Comment:
I like this but it's not clear to me why those are skipped. Is it because of our test logic is not compatible i.e. it expect different objects than would be produced when expressions flavour router is used?

Do you mind adding a comment about this to this func?

Reply:
added comments.